src/helpers/immutable.ts

```diff
@@ -0,0 +1,24 @@
+export function removeFromArray<T>(arr: Array<T>, index: number): Array<T> {
+    return [...arr.slice(0, index), ...arr.slice(index + 1)];
+}
+
+export function replaceItemInArray<T>(arr: Array<T>, element: T, index: number): Array<T> {
+    return [...arr.slice(0, index), element, ...arr.slice(index + 1)];
+}
+
+export function updateArrayItem<T>(arr: Array<T>, index: number, cb: (item: T) => T): Array<T> {
+    return [...arr.slice(0, index), cb(arr[index]), ...arr.slice(index + 1)];
+}
+
+export function insertInArray<T>(arr: Array<T>, element: T, index: number): Array<T> {
```

Comment:
This is generic insert function which is nice, but I think for lists we only need `push` like functionality, that is appending an element to the end.

Reply:
yes, it was written in another project originally just copied it over here.